libs/shared/flagd-core/src/lib/targeting/fractional.ts

```diff
@@ -0,0 +1,97 @@
+import {flagdPropertyKey, flagKeyPropertyKey, targetingPropertyKey} from "./common";
+import MurmurHash3 from "imurmurhash";
+
+export const fractionalRule = "fractional";
+
+export function fractional(data: unknown, context: Record<any, any>): string | null {
+  if (!Array.isArray(data)) {
+    return null;
+  }
+
+  const args = Array.from(data);
+  if (args.length < 2) {
+    console.error('Invalid targeting rule. Require at least two buckets.')
+    return null;
+  }
+
+  const flagdProperties = context[flagdPropertyKey];
+  if (!flagdProperties) {
+    return null;
+  }
+
+  let bucketBy: string;
+  let buckets: unknown[];
+
+  if (typeof args[0] == "string") {
+    bucketBy = args[0];
+    buckets = args.slice(1, args.length)
+  } else {
+    bucketBy = context[targetingPropertyKey];
+    if (!bucketBy) {
+      return null;
```

Comment:
Should we log before returning? It seems like this condition would only happen if something very unexpected happened.

Reply:
Added a log and yes this is if `targetingKey` is missing as a fallback option